python/ray/data/_internal/block_builder.py

```diff
@@ -18,6 +19,37 @@ def add_block(self, block: Block) -> None:
         """Append an entire block to the block being built."""
         raise NotImplementedError
 
+    def append_columns(self, block: Block, exprs: Dict[str, Expr]) -> Block:
```

Comment:
Expressions are abstractions having nothing to do with the block, right? 

Also, this method needs to be moved into TableBlockBuilder and we'd need to refactor that to support appending native column. 

Sorry, made a wrong judgement call that this would be relatively easy addition, but it now seems like it will actually require a refactor of the BlockBuilder. Let's avoid bloating this PR with it, so let's just keep this method  where it was and move it into BB later